### PR TITLE
Optimize some Opcodes

### PR DESCRIPTION
**Pull request by shargon**
- Optimized: STLOC, LDLOC, STARG, LDARG
- Close https://github.com/neo-project/neo-vm/issues/375

### CODE REVIEW

**Review by ProDog — APPROVED** [review]
Tested with NEP5 contract, no problem.

**Review by Tommo-L — APPROVED** [review]

File: src/Neo.Compiler.MSIL/Optimizer/Parser_OptimizeOpcode.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                switch (ins.OpCode)
+                {
+                    // Reduce unused data
+
+                    case OpCode.STLOC:
+                    case OpCode.LDLOC:
+                    case OpCode.STARG:
+                    case OpCode.LDARG:
+                        {
+                            if (items[x] is NefInstruction p1 && p1.Data.Length == 1)
```
Comment (Qiao-Jin) [inline src/Neo.Compiler.MSIL/Optimizer/Parser_OptimizeOpcode.cs:41]: Why check the type twice?

Comment (shargon) [reply src/Neo.Compiler.MSIL/Optimizer/Parser_OptimizeOpcode.cs]: done

**Comment by shargon** [comment]
Merge?

**Comment by shargon** [comment]
@lightszero could you review it please?

**Comment by shargon** [comment]
Merge?

File: src/Neo.Compiler.MSIL/Optimizer/Parser_OptimizeOpcode.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                    case OpCode.STLOC:
+                    case OpCode.LDLOC:
+                    case OpCode.STARG:
+                    case OpCode.LDARG:
+                        {
+                            if (ins.Data.Length == 1 && ins.Data[0] >= 0 && ins.Data[0] <= 6)
+                            {
+                                ins.SetOpCode((OpCode)(ins.OpCode - 7 + ins.Data[0]));
+                                ins.SetData(Array.Empty<byte>());
+                            }
```
Comment (erikzhang) [inline src/Neo.Compiler.MSIL/Optimizer/Parser_OptimizeOpcode.cs:45]: Why not use the correct opcode at the beginning?

**Pull request closed without merging**